region/maridia/inner-pink/Halfie Climb Room.json

```diff
@@ -502,57 +453,50 @@
       "link": [2, 1],
       "name": "Cross Room Jump",
       "notable": false,
+      "entranceCondition": {
+        "comeInJumping": {
+          "speedBooster": "any",
+          "minTiles": 1
+        }
+      },
       "requires": [
-        "canCrossRoomJumpIntoWater",
-        {"adjacentRunway": {
-          "fromNode": 2,
-          "usedTiles": 1.5,
-          "inRoomPath": [2, 1],
-          "physics": ["air"]
-        }}
+        "canCrossRoomJumpIntoWater"
       ],
-      "note": "Only requires a runway of approximately 1.5 tiles in the adjacent room."
+      "note": "Only requires a runway of 1 tile in the adjacent room.",
+      "devNote": "This can be done with a shorter runway (closed end) and canMomentumConservingTurnaround, but that type of connection doesn't exist."
     },
     {
       "link": [2, 1],
-      "name": "Cross Room Jump with Turnaround",
+      "name": "Suitless Water Shinecharge",
       "notable": false,
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 3,
+          "openEnd": 0
+        }
+      },
       "requires": [
-        "canCrossRoomJumpIntoWater",
-        "canMomentumConservingTurnaround",
-        {"adjacentRunway": {
-          "fromNode": 2,
-          "usedTiles": 0.5,
-          "inRoomPath": [2, 1],
-          "physics": ["air"]
-        }}
+        "canWaterShineCharge",
+        {"canShineCharge": {
+          "usedTiles": 33,
+          "openEnd": 2
+        }},
```

Comment:
What does this mean here? 33+2 is going to be always true.
Then there is `comeInShinecharging`  which does the real comparison with your usedTiles count.

Reply:
Looks like it doesnt mean anything here - I'll delete it.

Sometimes i think it's used to show something like `canShinecharge`, which might be better to add. Or simply adding `SpeedBooster`. It's used all over the place, like in that X-Mode strat you were reviewing above.

Reply:
The `canShineCharge` requirement implicitly includes a SpeedBooster item requirement. So as long as the SpeedBooster requirement is covered (e.g. in this case by the `canWaterShineCharge`), I think it would be fine to start dropping these 33+2 canShineCharge requirements.